server/media/media.go

```diff
@@ -70,14 +70,41 @@ func matchCORSOrigin(allowed []string, origin string) string {
 	}
 
 	if allowed[0] == "*" {
-		return "*"
+		if origin == "" {
+			return "*"
+		}
+		return origin
 	}
 
-	origin = strings.ToLower(origin)
-	for _, val := range allowed {
-		if strings.ToLower(val) == origin {
+	origin = strings.TrimSpace(strings.ToLower(origin))
+	for _, pattern := range allowed {
+		pattern = strings.TrimSpace(strings.ToLower(pattern))
+		if pattern == origin {
 			return origin
 		}
+
+		if strings.Contains(pattern, "*") {
+			parts := strings.SplitN(pattern, "://", 2)
+			if len(parts) != 2 {
+				continue
+			}
+			protocol, domain := parts[0], parts[1]
```

Comment:
Please use https://pkg.go.dev/net/url#ParseRequestURI instead of inventing your own.